### PR TITLE
fix(rds): default username 'admin' doesn't work for PostgreSQL

### LINKED ISSUE #10579: [rds] invalid default username when using Aurora PostgresSql
We recently added a property `credentials` to represent the master username and password.
It's an optional property with a default value of `admin` for the username.

### Reproduction Steps

```ts
const cluster = new rds.DatabaseCluster(this, 'Database', {
      engine: rds.DatabaseClusterEngine.AURORA_POSTGRESQL,
      instanceProps: {
        instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.MEDIUM),
        vpc,
        vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE },
      },
      defaultDatabaseName: 'database123',
      parameterGroup: rds.ParameterGroup.fromParameterGroupName(this, 'ParameterGroup', 'default.aurora-postgresql11'),
      instances: 1,
      s3ImportBuckets: [importBucket], 
    });
```

### What did you expect to happen?

Successful deployment when using default property for master username and password.
Specifying 

### What actually happened?

```console
MasterUsername admin cannot be used as it is a reserved word used by the engine
(Service: AmazonRDS; Status Code: 400; Error Code: InvalidParameterValue; Request ID: 37defeec-5b54-4541-ac05-6b5e92412ab2; Proxy: null)
```

### Environment

  - **CLI Version      : 1.63.0**
  - **Framework Version:**
  - **Node.js Version:** v14.8.0
  - **OS               :** Mac OS 10.15.6
  - **Language (Version):** TypeScript (3.8.3) 

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by skinny85**
In #10458, we started to default the master user name to 'admin'.
As it turns out, that actually doesn't work with PostgreSQL,
as 'admin' is a reserved word there.
Add a new optional property to IEngine called defaultUsername
that allows overriding the global 'admin' default on a per-Engine basis.
Set it to 'adminuser' for the Aurora Postgres Cluster engine and the Postgres Instance engine.

Fixes #10579

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by shivlaks — APPROVED** [review]
looks good to me! 
also glanced through the README to make sure we don't have any examples that explicitly try to use `admin` with PostgreSql and we're good there.

added a `do-not-merge` so @njlynch can also have a look.

File: packages/@aws-cdk/aws-rds/lib/cluster-engine.ts [hunk 1]
```diff
@@ -495,6 +495,7 @@ class AuroraPostgresClusterEngine extends ClusterEngineBase {
   private static readonly S3_EXPORT_FEATURE_NAME = 's3Export';
 
   public readonly engineFamily = 'POSTGRESQL';
+  public readonly defaultUsername = 'adminuser';
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/cluster-engine.ts:498]:
```suggestion
  public readonly defaultUsername = 'postgres';
```

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 2]
```diff
@@ -430,7 +430,7 @@ export interface DatabaseClusterProps extends DatabaseClusterBaseProps {
   /**
    * Credentials for the administrative user
    *
-   * @default - A username of 'admin' and SecretsManager-generated password
+   * @default - A username of 'admin' (except for Postgres, which uses 'adminuser') and SecretsManager-generated password
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:433]:
```suggestion
   * @default - A username of 'admin' (or 'postgres' for PostgreSQL) and SecretsManager-generated password
```

File: packages/@aws-cdk/aws-rds/lib/engine.ts [hunk 3]
```diff
@@ -39,4 +39,13 @@ export interface IEngine {
    * @default - the engine doesn't belong to any family
    */
   readonly engineFamily?: string;
+
+  /**
+   * The default name of the master database user if one was not provided explicitly.
+   * If this is `undefined`,
+   * the global default of 'admin' will be used,
+   * but that does not work for all the database engines -
+   * most notably, it's a reserved word in PostgreSQL.
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/engine.ts:48]:
```suggestion
   * The gobal default of 'admin' will be used if this is `undefined`.
   * Note that 'admin' is a reserved word in PostgreSQL and cannot be used.
```

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts [hunk 4]
```diff
@@ -578,6 +578,8 @@ export interface PostgresInstanceEngineProps {
  * The instance engine for PostgreSQL.
  */
 class PostgresInstanceEngine extends InstanceEngineBase {
+  public readonly defaultUsername = 'adminuser';
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:581]:
```suggestion
  public readonly defaultUsername = 'postgres';
```

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 5]
```diff
@@ -902,7 +902,7 @@ export interface DatabaseInstanceProps extends DatabaseInstanceSourceProps {
   /**
    * Credentials for the administrative user
    *
-   * @default - A username of 'admin' and SecretsManager-generated password
+   * @default - A username of 'admin' (except for Postgres, which uses 'adminuser') and SecretsManager-generated password
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/instance.ts:905]:
```suggestion
   * @default - A username of 'admin' (or 'postgres' for PostgreSQL) and SecretsManager-generated password
```

File: packages/@aws-cdk/aws-rds/test/test.cluster.ts [hunk 6]
```diff
@@ -1394,6 +1394,29 @@ export = {
...
+      engine: DatabaseClusterEngine.auroraPostgres({
+        version: AuroraPostgresEngineVersion.VER_9_6_12,
+      }),
+      instanceProps: { vpc },
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::SecretsManager::Secret', {
+      GenerateSecretString: {
+        SecretStringTemplate: '{"username":"adminuser"}',
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/test/test.cluster.ts:1413]:
```suggestion
        SecretStringTemplate: '{"username":"postgres"}',
```

File: packages/@aws-cdk/aws-rds/test/test.instance.ts [hunk 7]
```diff
@@ -997,6 +997,25 @@ export = {
...
+      vpc,
+      engine: rds.DatabaseInstanceEngine.postgres({
+        version: rds.PostgresEngineVersion.VER_9_5_7,
+      }),
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::SecretsManager::Secret', {
+      GenerateSecretString: {
+        SecretStringTemplate: '{"username":"adminuser"}',
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/test/test.instance.ts:1012]:
```suggestion
        SecretStringTemplate: '{"username":"postgres"}',
```

**Review by njlynch — COMMENTED** [review]
The RDS Console uses `postgres` as the default username for PostgreSQL databases; I think standardizing on that makes sense.

File: packages/@aws-cdk/aws-rds/lib/engine.ts (hunk 3 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-rds/lib/engine.ts:48]: ^^ typo on `global` in the suggestion

Comment (njlynch) [reply packages/@aws-cdk/aws-rds/lib/engine.ts:48]: Thanks @shivlaks ; fixed.

**Pull request merged**